src/osx/setup-env.cpp

```diff
@@ -9,17 +9,47 @@
 
 #include "filesystem.h"
 
+namespace {
+/// Set the environment variable name with the path value, if the path exists, and
+/// log whether successful or not.
+void setenvPathIfExists(const char* name, fs::path const& path, bool override) {
+    if (fs::exists(path)) {
+        setenv(name, path.c_str(), override);
+        std::cerr << "Overriding env var " << name << ": " << path << "\n";
+    } else {
+        std::cerr << "Skip override env var " << name << ", path doesn't exist: " << path << "\n";
+    }
+}
+
+}  // namespace
 
 void setupEnvironment() {
     /**
      * see https://gitlab.gnome.org/GNOME/gtk-mac-bundler/-/blob/master/examples/gtk3-launcher.sh
      * and https://gitlab.gnome.org/GNOME/gtk-mac-bundler/-/issues/12
      */
+
+    // Get the parent of the folder containing the application executable
     auto base = Stacktrace::getExePath().parent_path();  // Xournal++.app/Contents or $HOME/gtk/inst
 
-    if (fs::exists(base / "Resources")) {  // app-bundle
+    /* We consider the application to launch from a bundle iff the parent
+     * directory is named "Contents" and it has a "Resources" folder.
+     * Note that checking the latter only is insufficient, e.g. it will falsely
+     * detect a bundle from the build folder, since the repository root has a
+     * folder named "resources"
+     */
+    const bool isInBundle = base.filename() == "Contents" && fs::exists(base / "Resources");
+    if (isInBundle) {
+        // app-bundle, e.g. Xournal++.app/Contents
+        // Install prefix is actually the Resources folder
         base = base / "Resources";
-    }  // Now base is Xournal++.app/Contents/Resources or $HOME/gtk/inst
+        std::cerr << "xournalpp is running from a bundle\n";
+    } else {
+        // Assume the "base dir" is the install prefix, corresponding to
+        // e.g., build folder or $HOME/gtk/inst
```

Comment:
I just realized there's a case missing here. To summarize:
* the bundle has the executable at `Contents/bin/xournalpp`.
* in master branch, a source build has the executable located at `$build/xournalpp.`
* (The missing case) for release-1.1, the executable in the build folder is placed at `$build/src/xournalpp`

So maybe a special case check for `src` as a parent name is also needed here.

Reply:
We could also only set `XDG_DATA_DIRS` if the application is in a bundle, since the bundle should include any themes that may be needed anyways.

Reply:
> I just realized there's a case missing here. To summarize:
> 
> * the bundle has the executable at `Contents/bin/xournalpp`.
> * in master branch, a source build has the executable located at `$build/xournalpp.`
> * (The missing case) for release-1.1, the executable in the build folder is placed at `$build/src/xournalpp`
> 
> So maybe a special case check for `src` as a parent name is also needed here.

I agree, that there is a missing case for release-1.1. Instead of `Contents/bin/xournalpp` you mean `Contents/MacOS/xournalpp`. It doesn't matter though.

Currently running the PR via `jhbuild run src/xournalpp` from within the build folder results in the messages:
```
xournalpp is running from some non-bundle path
Skip override env var XDG_CONFIG_DIRS, path doesn't exist: "/Users/xournal-dev/xournalpp/build/etc/xdg"
Skip override env var XDG_DATA_DIRS, path doesn't exist: "/Users/xournal-dev/xournalpp/build/share"
Skip override env var GTK_IM_MODULE_FILE, path doesn't exist: "/Users/xournal-dev/xournalpp/build/lib/gtk-3.0/3.0.0/immodules.cache"
Skip override env var GDK_PIXBUF_MODULE_FILE, path doesn't exist: "/Users/xournal-dev/xournalpp/build/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache"
** Message: 16:51:11.404: Continue with GDK_PIXBUF_MODULE_FILE = (null)
** Message: 16:51:11.408: Setting LANG and LC_MESSAGES to de_CH.UTF-8
** Message: 16:51:11.613: TEXTDOMAINDIR = (null), Platform-specific locale dir = /Users/xournal-dev/xournalpp/build/share/locale, chosen directory = /Users/xournal-dev/xournalpp/build/share/locale
** Message: 16:51:11.649: Plugin "MigrateFontSizes" UI initialized

However, even if you remove the `/src` from those paths, they don't exist. So the approach seems flawed. I think if the app is running from a local build, it should still use the prefix `$HOME/gtk/inst`. 
```

Reply:
Does `jhbuild` set any environment variables when it runs the command?

Reply:
Yes, it sets a ton of environment variables. This can already been seen by executing `jhbuild run env` versus `env`.
In particular `XDG_DATA_DIRS`, `XDG_CONFIG_DIRS` are set and the `PATH` has several additions. It's 81 variables vs 19 variables in one test.